compiler/optimizations/gpuTransforms.cpp

```diff
@@ -79,7 +79,16 @@ static void markGPUSuitableLoops();
 
 
 static bool isDefinedInTheLoop(Symbol* sym, CForLoop* loop) {
-  return LoopStmt::findEnclosingLoop(sym->defPoint) == loop;
+  LoopStmt* curLoop = LoopStmt::findEnclosingLoop(sym->defPoint);
```

Comment:
This isn't new for this PR, but this algorithm is going to be O(n^2). Looking at the implementation of findEnclosingLoop it recursively walks up the AST to find the next enclosing loop.

Reply:
Isn't it O(n) where `n` is the depth of the expression? It'll jump to parents until you've found a loop. Stop there and check. Then, keep jumping to its parents, and not the original defPoint.

In any case, the `n` here should be very small, right?

Reply:
`findEnclosingLoop` is O(n) but we call `isDefinedInTheLoop` for each symbol in the loop right? In practice I think you're right the numbers here are small enough that there's not likely to be a performance problem.

Reply:
Ah, I see what you mean better now. Thanks for the clarification.

Yes, you're right. We probably want to do a top-down search on the loop body to find defpoints inside, rather than doing it symbol by symbol. Now that I understand your concern better, I am a little less optimistic about the performance implications. There may be tons of symbols, and repeating that for every forall can be noticeable for a large application. Let's keep this in mind in case we start to see long compilation times.